### PR TITLE
Fixes #33252 - Add Content View Module Stream Filter Page

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

File: webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    __('Name'),
+    __('Stream'),
+    __('Version'),
+    __('Context'),
+    __('Status'),
+  ];
+
+  const buildRows = useCallback((results) => {
+    const newRows = [];
+    const filterRules = filterResults.find(({ id }) => id === Number(filterId))?.rules || [];
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js:60]: Try `filterResults?.find` 

Sometimes refreshing the page fails on this. Even better would be to use the useSelector(state => selectCVFiltersStatus(state, cvId)); and then check if the status is not pending.

Something like this: https://github.com/Katello/katello/blob/31ea5f8cf3364dcced2f899d1e0aa30fbacbe845/webpack/scenes/ContentViews/Details/ComponentContentViews/ContentViewComponents.js#L188

File: webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        { title: context },
+        { title: <AddedStatusLabel added={added} /> },
+      ];
+
+      /* eslint-disable camelcase */
+
+      newRows.push({
+        cells,
+        module_stream_id: id,
+        packagefilterId: filterRules?.find(({ id }) => id === id)?.id, // eslint-disable-line max-len, no-shadow, no-self-compare
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js:87]: This needs to look for module_stream_id equal to module_strem.id..
Basically this:
`filterRules?.find(({ module_stream_id }) => module_stream_id === id)?.id`

Comment (sjha4) [reply webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js:87]: also, could you rename the variable to something like moduleStreamRuleId..

File: webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                </SplitItem>
+              </Split>
+            }
+          />
+        </div>
+      </Tab>
+      {(repositories.length || showAffectedRepos) &&
+      <Tab eventKey={1} title={<TabTitleText>{__('Affected Repositories')}</TabTitleText>}>
+        <div className="tab-body-with-spacing">
+          <AffectedRepositoryTable cvId={cvId} filterId={filterId} repoType="yum" setShowAffectedRepos={setShowAffectedRepos} />
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js:209]: We need this effect for the functionality where selecting to apply to a subset of repos takes user to the tab etc..You'll find this in other filter pages..We might need to refactor this somehow but for now, adding this is needed.
```
useEffect(() => {
    if (!repositories.length && showAffectedRepos) {
      setActiveTabKey(1);
    } else {
      setActiveTabKey(0);
    }
  }, [showAffectedRepos, repositories.length]);
```

File: webpack/scenes/ContentViews/Details/Filters/__tests__/cvModuleStreamFilter.test.js [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+  assertNockRequest(autocompleteScope);
+  assertNockRequest(cvFilterScope);
+  assertNockRequest(cvFiltersScope);
+  assertNockRequest(cvFiltersRuleBulkAddScope);
+  assertNockRequest(cvRequestCallbackScope);
+
+  assertNockRequest(moduleStreamsScope, done);
+});
+
+test('Can show affected repository tab and remove affected repos', async (done) => {
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Filters/__tests__/cvModuleStreamFilter.test.js:325]: We test this and the next test in PackageGroup filters test and it needs some changes based on Jeremy's PR for multipage select. We can safely remove these here cause the component for affected repository tab itself is tested in that test.

**Comment by chris1984** [comment]
@sjha4 updated, let me know how this looks?

**Review by sjha4 — APPROVED** [review]
APJ 🥳

**Comment by chris1984** [comment]
[test katello]

**Pull request merged**